README.md
# DerpleBuild
# MQ2TestBuild


Release/Changes.txt
Sep 20 2017 by eqmule
- Updated for LIVE

There will be no more public mq2 updates until I catch the guy that released the mq2 for agnarr.

send me a pm or skype text: eq.mule
if you have information. you can be anonymous.

longtime friends and loyal users here that I know and trust will be given the updates by me only from now on. (at no charge of course.) mq2 is still open source and free, its just that equistructs.h eqgame.h and eqdata.h wont be distributed or updated by me anymore. those change every patch and everything prior to this day is under gpl and can of course be requested or downloaded, but from now on until this crap is resolved, I have my own NEW includes that are NOT under the gpl.

I'm sick of having my hard work being taken and passed of by some leech on a server that I specifically requested we stay away from. I wont tolerate it.

mq2 is free and staying off of truebox was the only price people had to pay, cant do that? no mq2 for anyone anymore.

wanna step in and update it? lol go for it, it must be "easy" if eqmule can do it in a couple hours... pfft... yeah right...

/eqmule (yeah I'm bitter)

19 Sep 2017 by rswiders
- Modified Math.Clamp to use the parameter list of (N, Min, Max). ie, will clamp the value N between Min and Max.
- Added ${Group.AvgHPs} guess what it returns? - eqmule

18 Sep 2017 by eqmule
- Fixed an infinite loop bug and a variable parse bug in Eval functions. -cred bug report: creamo
- Eval functions that take multiple arguments MUST separate those with comma(s) from now on
  Example: (run with a cleric, war or shaman)
  [code]
  Sub Main
	/declare c_argueString string outer shm clr war
	/declare ChatSender string local ${Me} 

	/if (!${checkbot["shm war clr", ${ChatSender}]}) { <-NOTE the COMMA
		/echo ${Time} Only Func3: FALSE, this should be TRUE
	} else {
		/echo ${Time} Only Func3: TRUE, this should be TRUE
	}
  /return

  Sub checkbot(selectedBots, chatSender)
    /declare botSelected bool local FALSE
	/echo ${Time} in Sub checkbot selectedBots=${selectedBots} chatSender=${chatSender}
    /if (${Select[${Me.Class.ShortName},${selectedBots}]}) /varset botSelected TRUE
  /return ${botSelected}
  [/code]

17 Sep 2017 by eqmule
- Fixed ${Me.GukEarned}, ${Me.MMEarned}, ${Me.RujEarned}, ${Me.TakEarned}, ${Me.MirEarned} and ${Me.LDoNPoints}
- Added a new TLO PointMercant it only have one member : Item which is a pPointMerchantItemType.
  The pPointMerchantitemType has the following members: Name, ItemID, Price, ThemeID, IsStackable, IsLore, RaceMask, ClassMask, bCanUse
  Usage:
  /echo ${PointMercant} returns true if the LDON Mercant window is open and FALSE if not.
  /echo ${PointMercant.Item[1].Price} OR /echo ${PointMercant.Item[Ebon Hammer].Price}
  returns the Price for index 1 or whatever index Ebon Hammer is in if you do it by name.
  etc.

15 Sep 2017 starring rswiders as "the captain", eqmule as "crew member number 6" and special guest star: eqholic
- Updated to handle the new SPA's
- Fixed a bug in /next not taking its argument into account for nested for loops.
- Fix for labels
- Fixed the CSIDLWND struct (the SIDL define...) so since we have never had this right it was time, but note that
  if your plugin uses stuff like: pwnd->SidlText and its NOT an actual CSidlScreenWnd then you need to change it to whatever window it actually is.
  I guess questions about this should be directed to me (eqmule)

plugins changed outside of core are attached here:
http://www.macroquest2.com/phpBB3/viewtopic.php?f=29&t=11451&p=174467#p174467
MQ2Rez

14 Sep 2017 by eqmule
- Updated for TEST
- Fixed ${Familiar[x].} it will return familiars and not illusions now...
-
- Fixed a bunch of window structs that we where completely misusing anyway.
  this could require some plugin changes, contact me for help if you get errors.
-
- Fix: Variables that are undeclared but supplied as paramters now default to an actual NULL or 0 not the string "NULL"
-
- Inspecting the Blessed Spiritstaff of the Heyokah will no longer ctd you.
- Added Math.Clamp ${Math.Clamp[Min, N, Max]} will clamp N between Min and Max. - cred derple
  Example:
  ${Math.Clamp[1, 15, 30]} => 15
  ${Math.Clamp[20, 15, 30]} => 20
  ${Math.Clamp[1, 15, 10]} => 10
  Practical usage:
  /bct ranger //stick id ${Target.ID} ${Math.Clamp[25,${Target.Distance},150]} 
  this will make your ranger stay put if he is already between 25-150 meters away
  otherwise it will move him at most 150 away and at least 25 away.

12 Sep 2017 by eqmule and eqholic
- All plugin sources for this zip are attached to this post:
  http://www.macroquest2.com/phpBB3/viewtopic.php?f=29&t=11451&p=174415#p174415
- Changed MQ2MoveUtils to save stuff faster.
- MQ2ChatWnd got a new tlo /echo ${ChatWnd.Title}
- MQ2ChatWnd got a new command /setchattitle
- MQ2Melee updated - it has flags for down and holyshits called downflag0-60 and holyflag0-60
  you SHOULD set those flags to 2 if you parse ANY macro variables.
  Example:
  [code]
  downflag0=2
  downshit0=/if (${Macro.Paused}) /echo ${Macro} is PAUSED!
  [/code]
- MQ2Hud now updated with a new type HUDTYPE_MACRO which is 16
  This means you can now set the type to any combination of 1 2 4 8 and 16
  see http://www.macroquest2.com/wiki/index.php/MQ2HUD for more info.
  Basically using a hud that tries to parse variables that are used in macros, before they are declared will fail.
  This new flag lets mq2hud know to not even try unless its set.
-
- Adding events from a macro will now also automatically /declare variable as outer if used.
  Example: #Event Burn              "[MQ2] |${BurnText}|"
  As you can see this event uses the variable ${BurnText} and it will therefor /declare it as well.
  This means you can remove /declare BurnText in your Sub Main since adding the event did it for you already.
-

MQ2Main/skills.h
/*****************************************************************************
    MQ2Main.dll: MacroQuest2's extension DLL for EverQuest
    Copyright (C) 2002-2003 Plazmic, 2003-2005 Lax

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License, version 2, as published by
    the Free Software Foundation.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
******************************************************************************/
/*
 * skills.h
 *
 * ShowEQ Distributed under GPL
 * http://seq.sourceforge.net/
 */
    "1H Blunt",                // 0
    "1H Slashing",             // 1
    "2H Blunt",                // 2
    "2H Slashing",             // 3
    "Abjuration",              // 4
    "Alteration",              // 5
    "Apply Poison",            // 6
    "Archery",                 // 7
    "Backstab",                // 8
    "Bind Wound",              // 9
    "Bash",                    // 10
    "Block",                   // 11
    "Brass Instruments",       // 12
    "Channeling",              // 13
    "Conjuration",             // 14
    "Defense",                 // 15
    "Disarm",                  // 16
    "Disarm Traps",            // 17
    "Divination",              // 18
    "Dodge",                   // 19
    "Double Attack",           // 20
    "Dragon Punch",            // 21
    "Dual Wield",              // 22
    "Eagle Strike",            // 23
    "Evocation",               // 24
    "Feign Death",             // 25
    "Flying Kick",             // 26
    "Forage",                  // 27
    "Hand To Hand",            // 28
    "Hide",                    // 29
    "Kick",                    // 30
    "Meditate",                // 31
    "Mend",                    // 32
    "Offense",                 // 33
    "Parry",                   // 34
    "Pick Lock",               // 35
    "1H Piercing",             // 36
    "Riposte",                 // 37
    "Round Kick",              // 38
    "Safe Fall",               // 39
    "Sense Heading",           // 40
    "Singing",                 // 41
    "Sneak",                   // 42 //anything above 100 means a successrate of 100%
    "Specialize Abjure",       // 43
    "Specialize Alteration",   // 44
    "Specialize Conjuration",  // 45
    "Specialize Divination",   // 46
    "Specialize Evocation",    // 47
    "Pick Pockets",            // 48
    "Stringed Instruments",    // 49
    "Swimming",                // 50
    "Throwing",                // 51
    "Tiger Claw",              // 52
    "Tracking",                // 53
    "Wind Instruments",        // 54
    "Fishing",                 // 55
    "Make Poison",             // 56
    "Tinkering",               // 57
    "Research",                // 58
    "Alchemy",                 // 59
    "Baking",                  // 60
    "Tailoring",               // 61
    "Sense Traps",             // 62
    "Blacksmithing",           // 63
    "Fletching",               // 64
    "Brewing",                 // 65
    "Alcohol Tolerance",       // 66
    "Begging",                 // 67
    "Jewelry Making",          // 68
    "Pottery",                 // 69
    "Percussion Instruments",  // 70
    "Intimidation",            // 71
    "Berserking",              // 72
    "Taunt",                   // 73
    "Frenzy",                  // 74
    "Remove Trap",             // 75
    "Triple Attack",           // 76
    "2H Piercing",             // 77
    "78",                      // 78
    "79",                      // 79
    "80",                      // 80
    "81",                      // 81
    "82",                      // 82
    "83",                      // 83
    "84",                      // 84
    "85",                      // 85
    "86",                      // 86
    "87",                      // 87
    "88",                      // 88
    "89",                      // 89
    "90",                      // 90
    "91",                      // 91
    "92",                      // 92
    "93",                      // 93
    "94",                      // 94
    "95",                      // 95
    "96",                      // 96
    "97",                      // 97
    "98",                      // 98
    "99",                      // 99
    "100",                     // 100
    "101",                     // 101
    "102",                     // 102
    "103",                     // 103
    "104",                     // 104
    "Harm Touch",              // 105
    "106",                     // 106
    "Lay Hands",               // 107
    "108",                     // 108
    "109",                     // 109
    "110",                     // 110
    "Slam",                    // 111
    "112",                     // 112
    "113",                     // 113
    "Inspect",                 // 114
    "Open",                    // 115
    "Reveal Trap",             // 116
    "117",                     // 117
    "118",                     // 118
    "119",                     // 119
    "120",                     // 120
    "121",                     // 121
    "122",                     // 122
    "123",                     // 123
    "124",                     // 124
    "125",                     // 125
    "126",                     // 126
    "127",                     // 127 this should be last one
    "128",                     // 128 here for compatibility
    "129",                     // 129 here for compatibility
    "130",                     // 130 here for compatibility
    "131",                     // 131 here for compatibility
    "Throw Stone",             // 132 here for compatibility 
